tests/Hedgehog.Linq.Tests/NameTests.cs

```diff
@@ -36,8 +37,10 @@ public static IEnumerable<object[]> AllPublicMembers()
                     }
 
                     var mi = member as MethodInfo;
```

Comment:
This could be turned into an "if pattern match"

```csharp
if (member is MethodInfo mi && ...)
```

Reply:
Thank you @adam-becker. If we can switch to C# 8.0 we can use recursive patterns and static local functions:

```c#
switch (member)
{
    // Ignore special members like indexers etc.
    case PropertyInfo { IsSpecialName: true }:
    case MethodInfo mi when NotOfInterest(mi):
        continue;
    default:
        // Avoid covariant conversion from MemberInfo[] to object[].
        yield return new object[] { member };
        break;
}

static bool NotOfInterest(MethodBase mi) =>
       mi.IsSpecialName
    || mi.IsConstructor
    || mi.Name.StartsWith("get_")
       // Static inline methods, by convention starting with '`'.
    || mi.Name.StartsWith("`");
```

Reply:
So we can specify the C# language version in the *.csproj*, e.g.:

```diff
diff --git a/tests/Hedgehog.Linq.Tests/Hedgehog.Linq.Tests.csproj b/tests/Hedgehog.Linq.Tests/Hedgehog.Linq.Tests.csproj
index e9833ba..3d8748c 100644
--- a/tests/Hedgehog.Linq.Tests/Hedgehog.Linq.Tests.csproj
+++ b/tests/Hedgehog.Linq.Tests/Hedgehog.Linq.Tests.csproj
@@ -2,6 +2,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>netcoreapp3.0;net461</TargetFrameworks>
+    <LangVersion>8.0</LangVersion>
     <ApplicationIcon />
     <OutputType>Library</OutputType>
     <StartupObject />
```

Reply:
https://github.com/hedgehogqa/fsharp-hedgehog/pull/285/commits/29e5293c4c9fe2292414720c817df9ac38172758